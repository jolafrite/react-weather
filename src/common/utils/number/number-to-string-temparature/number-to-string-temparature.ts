import { toFahrenheit } from "../to-fahrenheit/to-fahrenheit";

export const numberToStringTemperature = (temperature: number) => {
  const celcius = Math.round(temperature)
  const fahrenheit = Math.round(toFahrenheit(temperature));

  return `${celcius}Â°/${fahrenheit}F`
};
