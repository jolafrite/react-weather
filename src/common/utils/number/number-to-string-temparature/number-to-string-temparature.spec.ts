import { numberToStringTemperature } from "./number-to-string-temparature";

describe('numberToStringTemperature', () => {

  test('should convert the number to a string <degree>Â° / <degree>F', () => {
    expect(
      numberToStringTemperature(20)
    ).toMatchSnapshot();
  });

  test('should round the temparatures', () => {
    expect(
      numberToStringTemperature(13.3333)
    ).toMatchSnapshot();
  });


});
