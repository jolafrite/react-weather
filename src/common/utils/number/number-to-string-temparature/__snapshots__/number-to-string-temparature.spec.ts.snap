// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`numberToStringTemperature should convert the number to a string <degree>° / <degree>F 1`] = `"20°C/68F"`;

exports[`numberToStringTemperature should round the temparatures 1`] = `"13°C/56F"`;
